relay-sampling/src/lib.rs

```diff
@@ -84,1522 +84,39 @@ pub use dsc::DynamicSamplingContext;
-            public_key: ProjectKey::parse("abd0f232775f45feab79864e580d160b").unwrap(),
-            release: Some("1.1.1".to_string()),
-            user: TraceUserContext {
-                user_segment: "vip".to_owned(),
-                user_id: "user-id".to_owned(),
-            },
-            replay_id: Some(Uuid::new_v4()),
-            environment: Some("debug".to_string()),
-            transaction: Some("transaction1".into()),
-            sample_rate: None,
-            sampled: None,
-            other: BTreeMap::new(),
-        };
-
-        for (rule_test_name, expected, condition) in conditions.iter() {
-            let failure_name = format!("Failed on test: '{rule_test_name}'!!!");
-            assert!(condition.matches(&dsc) == *expected, "{failure_name}");
-        }
-    }
-
-    #[test]
-    /// test various rules that do not match
-    fn test_does_not_match() {
-        let conditions = [
-            (
-                "release",
-                and(vec![
-                    glob("trace.release", &["1.1.2"]),
-                    eq("trace.environment", &["debug"], true),
-                    eq("trace.user", &["vip"], true),
-                ]),
-            ),
-            (
-                "user segment",
-                and(vec![
-                    glob("trace.release", &["1.1.1"]),
-                    eq("trace.environment", &["debug"], true),
-                    eq("trace.user", &["all"], true),
-                ]),
-            ),
-            (
-                "environment",
-                and(vec![
-                    glob("trace.release", &["1.1.1"]),
-                    eq("trace.environment", &["prod"], true),
-                    eq("trace.user", &["vip"], true),
-                ]),
-            ),
-            (
-                "transaction",
-                and(vec![
-                    glob("trace.release", &["1.1.1"]),
-                    glob("trace.transaction", &["t22"]),
-                    eq("trace.user", &["vip"], true),
-                ]),
-            ),
-        ];
-
-        let dsc = DynamicSamplingContext {
-            trace_id: Uuid::new_v4(),
-            public_key: ProjectKey::parse("abd0f232775f45feab79864e580d160b").unwrap(),
-            release: Some("1.1.1".to_string()),
-            user: TraceUserContext {
-                user_segment: "vip".to_owned(),
-                user_id: "user-id".to_owned(),
-            },
-            replay_id: Some(Uuid::new_v4()),
-            environment: Some("debug".to_string()),
-            transaction: Some("transaction1".into()),
-            sample_rate: None,
-            sampled: None,
-            other: BTreeMap::new(),
-        };
-
-        for (rule_test_name, condition) in conditions.iter() {
-            let failure_name = format!("Failed on test: '{rule_test_name}'!!!");
-            assert!(!condition.matches(&dsc), "{failure_name}");
-        }
-    }
-
-    #[test]
-    fn test_partial_trace_matches() {
```

Comment:
removed because I don't want to test specific Getters in this crate, maybe i should keep it if we don't want to move `DynamicSamplingContext` ?